boards/shields/nrf7002_ek/nrf7002_ek.overlay

```diff
@@ -4,6 +4,17 @@
  */
 #include <freq.h>
 
+/ {
```

Comment:
I think it would be valuable to utilize the generic pin forwarder. We could then get rid of `CONFIG_MPSL_CX_PIN_FORWARDER` and all the code associated with it.

Since this overlay should be applicable to both nRF52 SoCs and nRF53 application core, we can neither leave only `nrf_radio_coex` node here nor replace `nrf_radio_coex` node with an entry to the generic forwarder. However, what if we added _both_ `nrf_radio_coex` node _and_ an entry to the generic forwarder? Only one of the nodes would actually be used in a given scenario - for nRF52, the generic forwarder would be unused, and for nRF53 the `nrf_radio_coex` node would be unused. That's not ideal, but in the end we would be using the generic forwarder on nRF53 and we'd have a single EK overlay for nRF52 and nRF53 application core.

What do you think @ahasztag @anangl?

Reply:
I see two problems with such approach:
- we describe exactly the same hardware twice (once with the coex node, the second time with the forwarder) which would be confusing for the user)
- If at some time in the future the WiFi node would reference the Coex node (in the way I have done it now, it does not do anything at the moment, but perhaps something like this could be needed in the future) it would be unclear which node should be referenced - the `nrf-radio-coex` node or the `pin-forwarder` node.